chromadb/test/property/test_embeddings.py

```diff
@@ -0,0 +1,178 @@
+import pytest
+import logging
+from hypothesis import given, assume, settings, note
+import hypothesis.strategies as st
+from typing import List, Set
+import chromadb
+from chromadb.api import API
+from chromadb.api.models.Collection import Collection
+from chromadb.test.configurations import configurations
+import chromadb.test.property.strategies as strategies
+import numpy as np
+import numpy
+from hypothesis.stateful import (
+    Bundle,
+    RuleBasedStateMachine,
+    rule,
+    initialize,
+    precondition,
+    consumes,
+    run_state_machine_as_test,
+    multiple,
+    invariant,
+)
+from collections import defaultdict
+import time
+import chromadb.test.property.invariants as invariants
+
+
+traces = defaultdict(lambda: 0)
+
+
+def trace(key):
+    global traces
+    traces[key] += 1
+
+
+def print_traces():
```

Comment:
For my understanding why are we bookkeeping how many times the functions were called?

Reply:
Purely to understand how Hypothesis works and gain confidence that all states are being visited appropriately. There are some failure modes where it doesn't.

This is something we'd only ever look at when debugging an individual failing test, they'll be silent when running `pytest` without the `-s` option.